internal/proxy/ui_requests.go

```diff
@@ -270,3 +271,31 @@ func (p *Proxy) processToken(tokenData []byte, requestLogEntry *log.Entry) (pci
 
 	return
 }
+
+// unidles Jenkins only if it is idled and returns the
+// state of the pod, the http status of calling unidle, and error if any
+func (p *Proxy) startJenkins(ns, clusterURL string) (state clients.PodState, code int, err error) {
+	// Assume pods are starting and unidle only if it is in "idled" state
+	code = http.StatusAccepted
+
+	state, err = p.idler.State(ns, clusterURL)
+	if err != nil {
+		return
+	}
+
+	if state == clients.Idled {
```

Comment:
I remember you saying that even if jenkins is idling then also we have three states
Running -> Intermediate -> Idled. Is that right?

If the state is Intermediate, it would be worth knowing if Jenkins is idling or un-idling. But let's do that as a different PR, WDYT?

Reply:
@kishansagathiya at present idler does not give us that information. We need to build that logic into idler which should return state like "starting", "running", "idling", "idled" which at the moment, it doesn't and thus can't be built into proxy.

Reply:
@kishansagathiya  https://github.com/fabric8-services/fabric8-jenkins-idler/issues/246